### PR TITLE
Make psutil optional

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

File: synapse/metrics/__init__.py [hunk 1]
```diff
@@ -68,9 +68,18 @@ def register_cache(self, *args, **kwargs):
...
+    try:
+        import psutil
+        process = psutil.Process()
+        process.memory_info().rss
+    except (ImportError, AttributeError):
+        logger.warn(
+            "psutil is not installed or incorrect version."
+            " Disabling memory metrics."
+        )
+        return
```
Comment (NegativeMjark) [inline synapse/metrics/__init__.py]: Maybe move the rest of the function inside the try block rather than returning here?

Comment (erikjohnston) [inline synapse/metrics/__init__.py]: I prefer to keep `try...except` as narrow as possible, to ensure we don't accidentally consume legitimate errors.

Comment (NegativeMjark) [inline synapse/metrics/__init__.py]: Fair enough

**Comment by NegativeMjark** [comment]
LGTM

**Comment by richvdh** [comment]
why was this necessary? It is undone by #4027.

**Pull request merged**